motionplan/ik/nloptInverseKinematics.go

```diff
@@ -86,6 +84,46 @@ func (ik *NloptIK) Solve(ctx context.Context,
 	//nolint: gosec
 	randSeed := rand.New(rand.NewSource(int64(rseed)))
 	var err error
+	mInput := &State{Frame: ik.model}
+
+	// Determine optimal jump values; start with default, and if gradient is zero, increase to 1 to try to avoid underflow.
+	jump := make([]float64, 0, len(seed))
+	seedTest := append(make([]referenceframe.Input, 0, len(seed)), seed...)
+	eePos, err := ik.model.Transform(seed)
+	if err != nil {
+		return err
+	}
+	mInput.Configuration = seed
+	mInput.Position = eePos
+	seedDist := solveMetric(mInput)
+	for i, testVal := range seedTest {
+		for jumpVal := defaultJump; jumpVal < 1; jumpVal *= 10 {
+			seedTest[i] = referenceframe.Input{testVal.Value + jumpVal}
```

Comment:
could make the calculation of the gradient a function that takes a jump value.  You could call it here as well as below in the loop when we are doing gradient descent